witness/src/Dscp/CommonCLI.hs

```diff
@@ -84,14 +99,15 @@ baseKeyParamsParser who = do
          help "Password of secret key."
 
 appDirParamParser :: Parser AppDirParam
-appDirParamParser = AppDirectorySpecific <$>
-                        (strOption $
-                        long "appdir" <>
-                        metavar "FILEPATH" <>
-                        help "Path to application folder. If not specified, \
-                             \OS-dependent folder for applications will be \
-                             \used, for instance '%APPDIR%/Disciplina'.") <|>
-                    pure AppDirectoryOS
+appDirParamParser = fmap AppDirectorySpecific . strOption $
+     long "appdir" <>
+     metavar "FILEPATH" <>
+     help "Path to application folder. If not specified, \
+          \OS-dependent folder for applications will be \
+          \used, for instance '%APPDIR%/Disciplina'."
+     -- Removed default 'AppDirectoryOS'
+     -- [Note default-cli-params]
+
```

Comment:
This leaves no way to set `AppDirectoryOS` overriding config.
(Actually, this parser should've never been optional, one should've explicitly passed some `--os-appdir` option to make this parser return `AppDirectoryOS`)